src/client/Microsoft.Identity.Client/Internal/ClientCredentialWrapper.cs

```diff
@@ -150,43 +72,132 @@ private void ValidateCredentialParameters(ApplicationConfiguration config)
                 AuthenticationType = ConfidentialClientAuthenticationType.SignedClientAssertionDelegate;
             }
 
-            if (AuthenticationType == ConfidentialClientAuthenticationType.None)
+            switch (AuthenticationType)
```

Comment:
this switch should probably also throw NotImplementedException() as a default case?